test/e2e/e2e_helpers_test.go

```diff
@@ -424,6 +422,59 @@ func sortNamespacedNames(nns []types.NamespacedName) {
 }
 
 func testDataPath(description ginkgo.GinkgoTestDescription) string {
-       path := filepath.Join(description.ComponentTexts...)
-       return strings.ReplaceAll(path, " ", "_")
+	path := filepath.Join(description.ComponentTexts...)
+	return strings.ReplaceAll(path, " ", "_")
+}
+
+func GetIstioObject(istio *istiov1beta1.Istio, namespace, name string) error {
+	return testEnv.Client.Get(context.TODO(), client.ObjectKey{
+		Namespace: namespace,
+		Name:      name},
+		istio)
+}
+
+func GetMixerlessTelemetryStatus(istio *istiov1beta1.Istio, namespace, name string) (*bool, error) {
+	err := GetIstioObject(istio, namespace, name)
+	if err != nil {
+		return nil, err
+	}
+	// query current state and return it
+	return istio.Spec.MixerlessTelemetry.Enabled, nil
+}
+
+func SetMixerlessTelemetryState(istio *istiov1beta1.Istio, newState *bool) error {
```

Comment:
actually all functions in this file could be unexported imho
```suggestion
func setMixerlessTelemetryState(istio *istiov1beta1.Istio, newState *bool) error {
```

Reply:
These are called from the test e2e_test.go, and need to be exported.

Reply:
same as above for all similar functions in this file